[meenulekha] iP

meenulekha opened the pull request:
# Duke
>  "Action is the foundational key to all success." – Pablo Picasso

### Duke empowers you to turn ideas into accomplishments effortlessly. It's:
* Text-based
* Effortless to learn
* LIGHTNING-FAST to use

### You can get it today! :

* download here
* double click it
* add your tasks
* and let it manage tasks for you!
 ### and it is.... FREE

### Features:
- [ ] Managing tasks
- [ ] Manage deadlines (coming soon...)
- [ ] Reminders (coming soon...)

### You can use it to practice java too!

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

meenulekha committed:
Level-0: Rename, Greet, Exit

meenulekha committed:
Level-1: Echo

meenulekha committed:
Level 2. Add, List

meenulekha committed:
Level 3. Mark as Done

meenulekha committed:
Level-4: ToDo, Event, Deadline

meenulekha committed:
A-TextUiTesting: Automated Text UI Testing

meenulekha committed:
Level-5: Handle Errors

meenulekha committed:
Level-6: Delete

meenulekha committed:
A-Enums: Use Enums

meenulekha committed:
Error fix

meenulekha committed:
Add Level-7 save

meenulekha committed:
Merge pull request #1 from meenulekha/branch-Level-7

Add Level-7 save

meenulekha committed:
Add duke.txt file

meenulekha committed:
Merge pull request #2 from meenulekha/branch-Level-7

Add duke.txt file

meenulekha committed:
Merge branch 'branch-Level-7'

meenulekha committed:
Add level-8 dates and times

meenulekha committed:
Merge pull request #3 from meenulekha/branch-Level-8

Branch level 8

meenulekha committed:
Add A-MoreOOP: Use More OOP

meenulekha committed:
Add  A-Packages

meenulekha committed:
Merge commit 'a6f73244a20539e20c2189973ebf597df34ef631'

meenulekha committed:
no message

joenus commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,52 @@
+
+    public int i = 0;
+
+    //constructor
```
Please use the javadoc comments

joenus commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,52 @@
+            i--;
+        }
+    }
+    public List<Task> getTasks() {
```
Inconsistent spacing.

joenus commented on src/main/java/duke/ToDo.java:
```diff
@@ -0,0 +1,24 @@
+        return "[T]" + super.toString();
+    }
+
+
```
Inconsistent spacing.

joenus commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,78 @@
+    public static Task fromFileString(String fileString){
+        String[] elements = fileString.split("\\s*\\|\\s*");
+        Boolean done = Integer.parseInt(elements[1]) == 1;
+        //for each type of task
```
```suggestion
        // For each type of task
```

Comments should be capitalized and should have a space.

joenus commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,47 @@
+            while((line = reader.readLine()) != null){
+                //adding the task from the file to be read
+                //index++;
+                //tasks.set(index,duke.Task.fromFileString(line));
```
Comments not describing the code should be removed.

joenus commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,47 @@
+    public Storage(String filepath){
+        this.filepath = filepath;
+    }
+    public void saveToFile(int size, ArrayList<Task> tasks){
```
```suggestion
    public void saveToFile(int size, ArrayList<Task> tasks) {
```

Missing space.

joenus commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,29 @@
+        return "[E]" + super.toString() + " (from: " + from + " to: " + to + ")";
+    }
+    @Override
+    public String getType(){
```
```suggestion
    public String getType() {
```

Missing space.

joenus commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,29 @@
+    public String getType(){
+        return "E";
+    }
+    public String toFileString(){
```
```suggestion
    public String toFileString() {
```

Missing space.

joenus commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+        this.input = input;
+    }
+
+    public void execute(TaskList tasks, Ui ui){
```
```suggestion
    public void execute(TaskList tasks, Ui ui) {
```

Missing space.

joenus commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+        String action = elems[0];
+        String[] dateFormats = {"dd/MM/yyyy", "MMM dd yyyy"};
+        int i = 0;
+        while(tasks.getTask(i)!= null){
```
```suggestion
        while(tasks.getTask(i)!= null) {
```

Missing space.

joenus commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+            i++;
+        }
+
+        switch (action){
```
```suggestion
        switch (action) {
```

Missing space.

joenus commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+                break;
+            case "delete":
+                int toDelete = Integer.valueOf(elems[1]) -1;
+                if(toDelete >=0 && toDelete< i){
```
```suggestion
                if(toDelete >=0 && toDelete< i) {
```

Missing space.

joenus commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+
+            case "unmark":
+                for(int k =0; k < i ; k++){
+                    if (tasks.getTask(k) != null){
```
```suggestion
                    if (tasks.getTask(k) != null) {
```

Missing space.

joenus commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+                break;
+
+            case "unmark":
+                for(int k =0; k < i ; k++){
```
```suggestion
                for(int k =0; k < i; k++) {
```

Spacing

joenus commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,43 @@
+                " What can I do for you?");
+        System.out.println(line);
+
+        while(obj.hasNextLine()){
```
```suggestion
        while(obj.hasNextLine()){
```

Missing space.

joenus commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+        return "[D]" + super.toString() + " (by: " + by + ")";
+    }
+    @Override
+    public String getType(){
```
```suggestion
    public String getType() {
```

Missing space.

joenus commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+        return "D";
+    }
+    @Override
+    public String toFileString(){
```
```suggestion
    public String toFileString() {
```

Missing space.

joenus commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+                    }
+                }
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println("[ ] " + elems[1] );
```
```suggestion
                System.out.println("[ ] " + elems[1]);
```

Spacing

joenus commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,78 @@
+    public void unMark(){
+        this.isDone = false;
+    }
+    public String toString(){
```
```suggestion
    public String toString() {
```

Missing space.

joenus commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,78 @@
+        this.isDone = true;
+    }
+    //function to unmark
+    public void unMark(){
```
```suggestion
    public void unMark() {
```

Missing space.

tayziyi commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,43 @@
+
+
+public class Duke {
+    private String FILE_PATH = System.getProperty("user.dir") + "/src/main/java/data/duke.txt";
```
Good to declare this variable to be static, final

tayziyi commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public void run() {
+        String line = "------------------------------";
```
Good to declare this variable as final

tayziyi commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,47 @@
+package duke;
+
+import java.io.*;
```
good practice to import only what you need and not everything

tayziyi commented on src/main/java/duke/ToDo.java:
```diff
@@ -0,0 +1,24 @@
+
+public class ToDo extends Task {
+
+
```
extra line spacing

tayziyi commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+                }
+                if(toDate != null & fromDate!=null) {
+                    Task e = new Event(fromto[0], fromDate, toDate);
+                    //lst[i] = e;
```
remove comments which does not describe anything

tayziyi commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(e.toString());
+                    i++;
+                    //saveToFile(i, lst);
```
remove comments

tayziyi reviewed (commented):
Overall looks good, remember to follow conventions stated by both me and previous reviewer.

meenulekha committed:
Add JUnit Tests

meenulekha committed:
Add Jar file

meenulekha committed:
Add Javadoc comments

meenulekha committed:
Edit to coding standard

meenulekha committed:
Add implementation of Find

meenulekha committed:
Merge branch 'master' into A-CodingStandard

meenulekha committed:
Merge pull request #4 from meenulekha/A-CodingStandard

Edit to coding standard

meenulekha committed:
Merge tag 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java

meenulekha committed:
Merge branch 'master' of https://github.com/meenulekha/ip

# Conflicts:
#	src/main/java/duke/Parser.java

meenulekha committed:
Merge tag 'A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Parser.java

meenulekha committed:
add checkstyle

meenulekha committed:
Add checkstyle

meenulekha committed:
Add GUI Level-10

meenulekha committed:
Merge branch 'Level-10'

meenulekha committed:
Refactor code for improved code quality

This commit includes several improvements to enhance code quality in the Parser class:
- Refactored switch-case statements for better readability.
- Introduced separate methods for handling different actions to improve code organization.
- Enhanced comments for clarity.
- Javadocs
- Removed unnecessary code.
- Changed variable names for easier understanding

These changes aim to make the code more maintainable and understandable.

meenulekha committed:
Merge pull request #6 from meenulekha/branch-A-CodeQuality

Refactor code for improved code quality

meenulekha committed:
Refactor Storage and Parser classes with improvements and assertions

Storage Class:
- Updated constructor to include an assertion for ensuring the 'filepath' parameter is not null.
- Added assertions in 'saveToFile' method to check that the 'tasks' parameter is not null.

Parser Class:
- Introduced assertions for parameter validation throughout the 'execute' method.
- Asserted that 'elems' and 'dateFormats' are not null in the 'execute' method.
- Enhanced assertion usage in various methods for better parameter validation.
- Improved the formatting and clarity of existing code.

These changes enhance code robustness, introduce better validation practices, and provide meaningful assertions for easier debugging.

meenulekha committed:
Merge pull request #8 from meenulekha/branch-A-Assertions

Refactor Storage and Parser classes with improvements and assertions

meenulekha committed:
Refactor TaskPriorityComparator for Improved Task Sorting

- Modify the TaskPriorityComparator to prioritize ToDo tasks correctly.
- Improve the comparison logic to handle sorting tasks based on dates.
- Ensure tasks without dates are handled appropriately in the comparison.

This change addresses the issue of tasks not being sorted chronologically when saved to a file.

meenulekha committed:
Fix minor bug in Parser class

meenulekha committed:
Minor bug fix in the fromFileString method in the Task class

meenulekha committed:
no message

meenulekha committed:
Update README.md

meenulekha committed:
Add Ui image

meenulekha committed:
Update README.md

meenulekha committed:
Update README.md

meenulekha committed:
Update README.md

meenulekha committed:
Update README.md

meenulekha committed:
Update README.md

meenulekha committed:
Update README.md

meenulekha committed:
Delete space

meenulekha committed:
Update README.md
